### ISSUE TITLE
mockito spy lost annotations from the spied instance class

### ISSUE THREAD

**Issue by cassiuscai** [mockito/mockito#204]
Hi， here is my problem
- 1.  `SomeAnnotation` annotation to mark an class 
```java
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface SomeAnnotation {
}
```

- 2.  `SomeClass` marked by `SomeAnnotation`
```java
@SomeAnnotation
public class SomeClass {
}
```

- 3. My Test Case
```test
@Test
    public void testSpy() {
        SomeClass someClass = new SomeClass();
        Assertions.assertThat(someClass.getClass().getAnnotation(SomeAnnotation.class)).isNotNull();
        SomeClass spyOnSomeClass = Mockito.spy(someClass);
        Assertions.assertThat(spyOnSomeClass.getClass().getAnnotation(AnAnnotation.class)).isNotNull();
    }
```

- 4. My test fail because of spyOnAnClass doesn't have the annotation `SomeAnnotation` that `SomeClass` have

**Comment by bric3** [comment]
Hi, thanks for the report. Actually this a known issue of CGLIB.

I forgot about it in #171, I'll look if we can do soemthing with bytebuddy on this front.

**Comment by bric3** [comment]
Indeed fixed by #171
